assets/src/hooks/useDetour.ts

```diff
@@ -26,35 +28,64 @@ export enum DetourState {
 }
 
 export const useDetour = ({ routePatternId, shape }: OriginalRoute) => {
-  const [state, setState] = useState<DetourState>(DetourState.Edit)
+  const [snapshot, send] = useMachine(createDetourMachine)
+
+  /*
+   * There's probably a better way to do this? Tags or maybe context?
+   * Tags seem more appropriate, but weird to manage Out-Of-Bounds state via tags.
+   * Maybe this entire API call could be moved to and managed by an actor
+   * combined with parallel child states within the state machine?
+   * -- https://stately.ai/docs/promise-actors */
+  const isFinishedDrawing = snapshot.matches({
+    "Detour Drawing": { Editing: "Finished Drawing" },
+  })
+  const isSharingDetour = snapshot.matches({ "Detour Drawing": "Share Detour" })
+  const isInFinishedDetourState = isFinishedDrawing || isSharingDetour
 
-  const [startPoint, setStartPoint] = useState<ShapePoint | null>(null)
-  const [endPoint, setEndPoint] = useState<ShapePoint | null>(null)
-  const [waypoints, setWaypoints] = useState<ShapePoint[]>([])
+  const firstWaypoint = snapshot.context.waypoints.at(0)
+  const lastWaypoint = snapshot.context.waypoints.at(-1)
+  // Lets also just assert that we're not operating on the same array element
+  const has2Waypoints = snapshot.context.waypoints.length >= 2
 
   const { result: finishedDetour } = useApiCall({
     apiCall: useCallback(async () => {
-      if (startPoint && endPoint) {
-        return fetchFinishedDetour(routePatternId, startPoint, endPoint)
-      } else {
+      if (!isInFinishedDetourState) {
         return null
       }
-    }, [startPoint, endPoint, routePatternId]),
+
+      /* Until we have "typegen" in XState,
+       * we need to validate these exist for typescript
+       *
+       * > [Warning] XState Typegen does not fully support XState v5 yet. However,
+       * > strongly-typed machines can still be achieved without Typegen.
+       * > -- https://stately.ai/docs/migration#use-typestypegen-instead-of-tstypes
+       */
+      if (
+        !has2Waypoints ||
+        firstWaypoint === undefined ||
+        lastWaypoint === undefined
+      ) {
+        return null
+      }
+
+      return fetchFinishedDetour(routePatternId, firstWaypoint, lastWaypoint)
+    }, [
+      isInFinishedDetourState,
+      firstWaypoint,
+      lastWaypoint,
+      has2Waypoints,
+      routePatternId,
+    ]),
   })
 
   const { result: nearestIntersection } = useNearestIntersection({
-    latitude: startPoint?.lat,
-    longitude: startPoint?.lon,
+    latitude: snapshot.context.waypoints.at(0)?.lat,
+    longitude: snapshot.context.waypoints.at(0)?.lon,
   })
 
   const detourShape = useDetourDirections(
-    useMemo(
-      () =>
-        [startPoint, ...waypoints, endPoint].filter(
-          (v): v is ShapePoint => !!v
-        ),
-      [startPoint, waypoints, endPoint]
-    ) ?? []
+    // Shouldn't need `useMemo` anymore because the state machine is tracking this ref
```

Comment:
Nice! 

We don't need the comment now though, do we?

Reply:
https://github.com/mbta/skate/compare/c9f4c6281fb4cf0f6787dfca9377e401a9155d30..9dd609fc2b042e8f816ff0ffd46d503c651073c0